### PR TITLE
[Quant][devs] Separated implementations for quantized & non-quantized tensors for the unsqueeze function

### PR DESCRIPTION
**Pull request by dzdang**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #71648

Summary: This PR is part of a series of PRs addressing https://github.com/pytorch/pytorch/issues/54150,
related to using dispatcher for calls to quantized backends as opposed to if/else conditionals.
This particular PR separates the calls to quantized & non-quantized backends for unsqueeze
using a dispatcher.

Test plan: Additional testing was not implemented to test this change because test cases in the existing test suite already make use of the squeeze function for various backends.

Differential Revision: [D33809041](https://our.internmc.facebook.com/intern/diff/D33809041)

### CODE REVIEW

File: aten/src/ATen/native/TensorShape.cpp [hunk 1]
```diff
@@ -2025,7 +2025,32 @@ Tensor _unsafe_view(const Tensor& self, IntArrayRef size) {
   return self.view(size);
 }
 
-static Tensor unsqueeze_sparse(Tensor const &self, int64_t dim /* should already be wrapped */) {
+Tensor unsqueeze_qtensor(const Tensor& self, int64_t dim) {
```
Comment (jerryzh168) [inline aten/src/ATen/native/TensorShape.cpp:2028]: can we merge this function into `unsqueeze_quantized`

**Review by jerryzh168 — APPROVED** [review]

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33809041).

**Review by jerryzh168 — APPROVED** [review]

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33809041).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33809041).

**Comment by dzdang** [comment]
@dzdang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33809041).

**Pull request closed without merging**